code/game/objects/items/whistle.dm

```diff
@@ -34,21 +34,24 @@
 
 
 /obj/item/whistle/proc/whistle_playsound(mob/user as mob)
-	if(TIMER_COOLDOWN_CHECK(src, COOLDOWN_WHISTLE_BLOW))
+	CHECK_TICK
+	if(TIMER_COOLDOWN_CHECK(user, COOLDOWN_WHISTLE_BLOW))
 		user.balloon_alert(user, "Catch your breath!")
 		return
 
 	user.visible_message(span_warning("[user] blows into [src]!"))
 	playsound(get_turf(src), 'sound/items/whistle.ogg', volume, 1)
 
-	if(TIMER_COOLDOWN_CHECK(src, COOLDOWN_WHISTLE_WARCRY))
+	if(TIMER_COOLDOWN_CHECK(user, COOLDOWN_WHISTLE_WARCRY))
 		to_chat(user, span_notice("You have to wait a little longer to rally your troops..."))
 	else
-		TIMER_COOLDOWN_START(src, COOLDOWN_WHISTLE_WARCRY, 15 SECONDS)
+		TIMER_COOLDOWN_START(user, COOLDOWN_WHISTLE_WARCRY, 15 SECONDS)
 		for(var/mob/living/carbon/human/human in get_hearers_in_view(warcryrange, user.loc))
-			human.emote("warcry")
+			if(!TIMER_COOLDOWN_CHECK(human, COOLDOWN_WHISTLE_WARCRIER))
+				human.emote("warcry")
+				TIMER_COOLDOWN_START(human, COOLDOWN_WHISTLE_WARCRIER, 15 SECONDS)
```

Comment:
Why arent you just putting it on the mob that blew the whistle

Reply:
The cooldown being on per person allows for two things
- People cannot stack whistle on different people to earrape
- People can use other emotes after the fact as they were forced to do warcry in the first place, allowing for the player to still do something even if someone pressed the funny button